fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -160,6 +162,56 @@ public void createCatalog(String type, String catalogName, String comment, Map<S
         }
     }
 
+    private void reCreatCatalog(Catalog catalog, Map<String, String> alterProperties, boolean isReplay) throws DdlException {
+        String catalogName = catalog.getName();
+        String type = catalog.getType();
+        CatalogConnector newConnector = null;
+
+        writeLock();
+        try {
+            Map<String, String> newProperties = new HashMap<>(catalog.getConfig().size() + alterProperties.size());
+            newProperties.putAll(catalog.getConfig());
+            newProperties.putAll(alterProperties);
+
+            LOG.info("Recreate catalog [{}] with properties [{}]", catalogName, newProperties);
+
+            newConnector = connectorMgr.createHiddenConnector(
+                    new ConnectorContext(catalogName, type, newProperties), isReplay);
+            if (null == newConnector) {
+                throw new DdlException("create hidden connector failed");
+            }
+
+            // drop old connector
+            connectorMgr.removeConnector(catalogName);
+
+            // replace old connector with new connector
+            connectorMgr.addConnector(catalogName, newConnector);
+
+            String serviceName = newProperties.get("ranger.plugin.hive.service.name");
```

Comment:
If user first alter `ranger.plugin.hive.service.name`, then alter other properties, the AccessControl will be reset to RangerStarRocksAccessController or NativeAccessController

Reply:
get